tests/test_core.py

```diff
@@ -96,7 +96,7 @@ def test_hashfile(tmpdir, algo, digest):
 def test_redirect_prefix_encodes_newlines():
     """Ensure raw newlines are url encoded in the generated redirect."""
     request = Namespace(
-        fullpath='/\nSet-Cookie:malicious=1;'
+        custom_fullpath='/\nSet-Cookie:malicious=1;'
```

Comment:
It should be changed, because it uses the `Namespace` instance not the `Bottle` instance.